File: core/src/main/java/com/lomekwi/cave/ui/listeners/ChangeListenerX.java
package com.lomekwi.cave.ui.listeners;

import com.badlogic.gdx.scenes.scene2d.Actor;
import com.badlogic.gdx.scenes.scene2d.utils.ChangeListener;

public class ChangeListenerX extends ChangeListener {
    private final Runnable runnable;
    public ChangeListenerX(Runnable runnable){
        this.runnable=runnable;
    }
    @Override
    public void changed(ChangeEvent event, Actor actor) {
        runnable.run();
    }
}

File: core/src/main/scala/com/lomekwi/cave/ui/listeners/ChangeListenerX.scala
package com.lomekwi.cave.ui.listeners

import com.badlogic.gdx.scenes.scene2d.Actor
import com.badlogic.gdx.scenes.scene2d.utils.ChangeListener

class ChangeListenerX(private val runnable: Runnable) extends ChangeListener {
  override def changed(event: ChangeListener.ChangeEvent, actor: Actor): Unit = {
    runnable.run()
  }
}
